ubenchmark/src/main/java/org/postgresql/benchmark/statement/ProcessBoolean.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.benchmark.statement;
+
+import org.postgresql.util.ConnectionUtil;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * Benchmark to test performance of ResultSet.getBoolean() method.
+ *
+ * @author jorsol
+ */
+@Fork(value = 1, jvmArgsPrepend = "-Xmx128m")
+@Measurement(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@State(Scope.Thread)
+@Threads(1)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class ProcessBoolean {
+
+  private Connection connection;
+  private PreparedStatement ps;
+  private ResultSet rs;
+
+  @Param({"5", "10", "50", "100", "10000"})
+  public int rowsize;
+
+  @Setup(Level.Trial)
+  public void setUp() throws SQLException {
+    Properties props = ConnectionUtil.getProperties();
+
+    connection = DriverManager.getConnection(ConnectionUtil.getURL(), props);
+    connection.createStatement().execute("create temp table booltab (a boolean)");
+    ps = connection.prepareStatement("insert into booltab values(random() > 0.5)");
+    for (int i = 0; i < rowsize; i++) {
+      ps.executeUpdate();
+    }
+    ps.close();
+
+    ps = connection.prepareStatement("select a from booltab", ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+    rs = ps.executeQuery();
+  }
+
+  @TearDown(Level.Trial)
+  public void tearDown() throws SQLException {
+    rs.close();
+    ps.close();
+    connection.close();
+  }
+
+  @Benchmark
+  public void getBoolean() throws SQLException {
+    rs.first();
+    while (rs.next()) {
+      rs.getBoolean(1);
```

Comment:
:confused: that test (ProcessResultSet) hammer the db executing the query every single time, it's not a realistic way to measure the java method performance since there is a lot overhead from the db. Anyway, I will add the BOOL to that test and add a "Blackhole" to ProcessBoolean to avoid the `dead code elimination`.